src/cmd/flux-account-service.py

```diff
@@ -0,0 +1,423 @@
+            print("registered accounting service", file=sys.stderr)
+        except FileExistsError:
+            logging.basicConfig(format="%(levelname)s:%(message)s", level=logging.ERROR)
+            logging.error("flux-accounting service is already registered")
+
+        # register signal watcher for SIGTERM to initiate shutdown
+        self.handle.signal_watcher_create(signal.SIGTERM, self.shutdown).start()
+        self.handle.signal_watcher_create(signal.SIGINT, self.shutdown).start()
+
+        endpoints = [
+            "view_user",
+            "add_user",
+            "delete_user",
+            "edit_user",
+            "view_bank",
+            "add_bank",
+            "delete_bank",
+            "edit_bank",
+            "view_job_records",
+            "update_usage",
+            "add_queue",
+            "view_queue",
+            "delete_queue",
+            "edit_queue",
+            "add_project",
+            "view_project",
+            "delete_project",
+            "shutdown",
+        ]
+
+        for name in endpoints:
+            self.handle.msg_watcher_create(
+                getattr(self, name), FLUX_MSGTYPE_REQUEST, f"accounting.{name}", self
+            ).start()
+
+    def shutdown(self, handle, watcher, signum, arg):
+        print("Shutting down...", file=sys.stderr)
+        self.conn.close()
+        self.handle.service_unregister("accounting").get()
+        self.handle.reactor_stop()
+
+    def view_user(self, handle, watcher, msg, arg):
+        try:
+            # call view-user function
+            val = u.view_user(self.conn, msg.payload["username"])
+
+            payload = {"view_user": val}
+
+            # handle a test-server.view_user request
+            handle.respond(msg, payload)
+        except sqlite3.OperationalError as db_err:
+            # an exception was raised in the view_user method
+            handle.respond_error(msg, db_err.errno, str(db_err))
+        except Exception as exc:
+            # fall through to a non-OSError exception
+            handle.respond_error(msg, 0, str(exc))
+
+    def add_user(self, handle, watcher, msg, arg):
+        try:
+            val = u.add_user(
+                self.conn,
+                msg.payload["username"],
+                msg.payload["bank"],
+                msg.payload["userid"],
+                msg.payload["shares"],
+                msg.payload["max_running_jobs"],
+                msg.payload["max_active_jobs"],
+                msg.payload["max_nodes"],
+                msg.payload["queues"],
+                msg.payload["projects"],
+            )
+
+            payload = {"add_user": val}
+
+            handle.respond(msg, payload)
+        except ValueError as val_err:
+            handle.respond_error(msg, val_err.errno, str(val_err))
+        except sqlite3.IntegrityError as integrity_err:
+            handle.respond_error(msg, integrity_err.errno, str(integrity_err))
```

Comment:
Yeah, actually, after looking through the `try-except` blocks in a lot of these functions, it turns out the exceptions being listed here are already handled in the bindings themselves, so I'm not sure I need them to be listed here. I think I can just handle the `KeyError` exception and a general `Exception` for any non-`OSError` not already handled by the bindings. So maybe I should just remove the various `ValueError`, `sqlite3.OperationalError` and `IntegrityError`, etc. from `flux-account-service.py`.

Perhaps a near PR in the future should consist of providing more context and clarity to the exceptions being handled in the bindings themselves (e.g the example you provided in your comment: `f"sqllite3 operational error: {op_err}"`). I can open a separate issue on this.

Reply:
I know pylint throws an error if you use general `Exception` in a try/except block, but you must have that suppressed here.
But yes, if the error messages in the exception are already well formatted, then all you have to do is something like

```python
    except (ValueError, OSError, sqlite3.Error) as exc:
        handle.respond_error(msg, 0, str(exc))
```

Reply:
Yeah, that seems right, though it seems more "pythonic" to just let the ValueError be raised to the caller?

Reply:
I would open a separate issue functions that return strings instead of raising an exception on error. This is probably ok for now, but it seems like the caller should be able to tell the difference between success and failure for these functions?

Reply:
Agreed. I can open a separate issue on converting the functions that return strings on error to instead raise the appropriate exception. That should probably be near-future work, if not work immediately following this PR. Does that sound ok?

Reply:
Yes, sounds good

Reply:
Cool. I force-pushed up a change to remove the `except` blocks for `Exception`s currently handled by the bindings.